frontend/src/useGetAuthorData.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import axios from "axios";
+import getCsrfToken from "./utils";
+
+// hook <- component but in this case, we don't return jsx (ie. any visual stuff, ui). We only return data and states
+// that we may need upon calling the hook.
+// installed axios to use this hook
+
+export default function useGetAuthorData(url) {
+    // do an api call whenever we call this hook.
+
+    const [authorID, setAuthorID] = useState(null)
+    const [data, setData] = useState(null);             // set as null because we don't really know the state the data is in initially.
+    const [loading, setLoading] = useState(false);      // boolean; set to false initially becuase nothing is loading yet till we call something to load 
+    const [error, setError] = useState(null);           // 
+
+//   useEffect(() => {                                     // insert url to get in touch with the api
+//     setLoading(true);                                   // set loading to true after a call to an api is made 
+//     axios
+//     .get(url)
+//     .then((response) => {
+//         setData(response.data);
+//     }).catch((error) => {                               // catch will be call if there's an error 
+//         setError(error);
+//     }).finally(() => {
+//         setLoading(false);                              // Loading phase is done when reached 
+//     });
+//   }, [url]);                                            // useEffect need dependency array which will have url 
+
+//   // console.log(data);
+//   return {data, loading, error};                        // return an object containing the three
+
+    useEffect(() => {

```

Comment:
Is it possible to directly use promises instead of async. You can use the `.then()` function.

```js
let promise1 = axios.get("/get_author_id");
let promise2 = promise1.then((response) => {
    axios.get("/authors/" + response.data["author_id"] + '/');
});

promise2.then((response) => {
    setAuthorData(response.data)
})
```

Something like this ^

Reply:
Hmmm that's what I have in my useFetch method but isn't using .then() with promises and async/await the same? Is there a particular reason of doing so? I don't really know a lot about the differences the two so I'm curious haha

Reply:
According to this [page](https://www.geeksforgeeks.org/difference-between-promise-and-async-await-in-node-js/)

> 2. Async/Await: Async/Await is used to work with promises in asynchronous functions. It is basically syntactic sugar for promises. It is just a wrapper to restyle code and make promises easier to read and use. It makes asynchronous code look more like synchronous/procedural code, which is easier to understand.

Do you still want me to switch async await to promises? @nevilkandathil

Reply:
No thats fine